### PR TITLE
Add candies to cart

### PR DESCRIPTION
**Pull request by fmachucas**
Adds custom events to intercommunicate apps.
* micro-candy-app throw event with selected candies data.
* stenciljs my-container webcomponent, listen to these event and log its content.

### CODE REVIEW

File: micro-candy-app/src/App.js [hunk 1]
```diff
@@ -24,34 +24,54 @@ const CandyInput = ({candy}) =>
...
+  addToCart(e) {
+    e.preventDefault();
+
+    const candies = {
+      lolypop: e.target.lolypop.value,
+      chocolate: e.target.chocolate.value
+    };
+
+    window.dispatchEvent(new CustomEvent('addToCart', { detail: candies }));
+    window.dispatchEvent(new CustomEvent('closeApp'));
```
Comment (roberto) [inline micro-candy-app/src/App.js:36]: :+1:

File: micro-candy-app/src/App.js [hunk 2]
```diff
@@ -24,34 +24,54 @@ const CandyInput = ({candy}) =>
...
+  }
+
+  cancel(e) {
+    e.preventDefault();
+    window.dispatchEvent(new CustomEvent('closeApp'));
+
```
Comment (roberto) [inline micro-candy-app/src/App.js:42]: [blank space](https://www.youtube.com/watch?v=e-ORhEE9VVg)

File: micro-candy-app/src/App.js [hunk 3]
```diff
@@ -24,34 +24,54 @@ const CandyInput = ({candy}) =>
...
 class App extends Component {
+  addToCart(e) {
+    e.preventDefault();
+
+    const candies = {
+      lolypop: e.target.lolypop.value,
+      chocolate: e.target.chocolate.value
+    };
+
+    window.dispatchEvent(new CustomEvent('addToCart', { detail: candies }));
```
Comment (roberto) [inline micro-candy-app/src/App.js:35]: I think we could pass which app is sending the event.
```
{
  details: candies,
  app: 'candy-store' // this should be a constant in the app too.
}
```

**Review by roberto — CHANGES_REQUESTED** [review]

**Review by roberto — APPROVED** [review]

File: micro-candy-app/src/App.js (hunk 3 above)
Comment (roberto) [reply micro-candy-app/src/App.js:35]: I'm going to add it in another PR. thx!

File: stenciljs/my-container/src/components/my-container/my-container.tsx [hunk 4]
```diff
@@ -29,11 +29,16 @@ export class MyContainer {
     this.activeApp = event.detail;
   }
 
-  @Listen('closeApp')
+  @Listen('window:closeApp')
   closeAppHandler() {
     this.activeApp = null;
   }
 
+  @Listen('window:addToCart')
```
Comment (rogeriochaves) [inline stenciljs/my-container/src/components/my-container/my-container.tsx:37]: 💯👌

**Pull request merged**